From 53482bd0e4effc9464bdad1026956cd0b5773a87 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@freescale.com>
Date: Wed, 18 Apr 2012 20:44:30 +0000
Subject: [PATCH 116/128] dpaa_eth: Adjust default buffer size

Default buffer size is based on the L2 MAXFRM value, such that
buffers in the pool are large enough to accomodate incoming frames
in a contiguous fd.

But while MAXFRM includes the FCS size, we don't need to account
for that in our buffers, since FCS is stripped down by the DTSEC.
Adjust default buffer size accordingly; it's a minor change, but
which better reflects reality and also helps a bit with the
recycling of termination buffers.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@freescale.com>
[Kevin: Original patch taken from FSL
QorIQ-SDK-V1.2-SOURCE-20120614-yocto.iso image]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
index 9180667..efa4c64 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.h
@@ -142,7 +142,11 @@ void fsl_dpaa_eth_set_hooks(struct dpaa_eth_hooks_s *hooks);
  */
 #define DEFAULT_BUF_SIZE	PAGE_SIZE
 #else
-#define DEFAULT_BUF_SIZE DPA_BP_SIZE(fsl_fman_phy_maxfrm);
+/*
+ * Default buffer size is based on L2 MAXFRM value, minus the FCS which is
+ * stripped down by hardware.
+ */
+#define DEFAULT_BUF_SIZE DPA_BP_SIZE(fsl_fman_phy_maxfrm - ETH_FCS_LEN)
 #endif /* CONFIG_DPAA_ETH_SG_SUPPORT */
 
 /*
-- 
1.7.9.7

